File: plugin/plugin/src/main/java/io/github/remote/konfig/RemoteKonfigPlugin.java
package io.github.remote.konfig;

import org.gradle.api.Plugin;
import org.gradle.api.Project;

public final class RemoteKonfigPlugin implements Plugin<Project> {
    @Override
    public void apply(Project target) {
        target.getPlugins().withId("com.android.application", plugin -> {
            Project apiProject = target.getRootProject().findProject(":api");
            if (apiProject == null) {
                throw new IllegalStateException("remote-konfig plugin requires a project named ':api'");
            }
            target.getDependencies().add("implementation", apiProject);
            target.getLogger().info("remote-konfig applied: added :api dependency to " + target.getPath());
        });
    }
}

File: plugin/plugin/src/main/kotlin/io/github/remote/konfig/RemoteKonfigPlugin.kt
package io.github.remote.konfig

import org.gradle.api.Plugin
import org.gradle.api.Project

class RemoteKonfigPlugin : Plugin<Project> {
    override fun apply(target: Project) {
        target.plugins.withId("com.android.application") {
            val apiProject = target.rootProject.findProject(":api")
                ?: error("remote-konfig plugin requires a project named ':api'")
            target.dependencies.add("implementation", apiProject)
            target.logger.info("remote-konfig applied: added :api dependency to ${target.path}")
        }
    }
}
